content/compute-app/non-functional-requirements/readme.md

```diff
@@ -0,0 +1,156 @@
+- Paradigm shift from network to identity as security boundary in cloud.
+
+## Operational excellence
+
+- **Pillar Concept** - monitoring, supportability requirements, DevOps and automation, serviceability, SLA monitoring, problem and incident management.
+
+- Solution complexity is an underrated consideration.
+  - Consider your team's size, workload, and level of skill. Will they actually be able to design, build/implement, monitor, and maintain this solution in practice? Or are you throwing them a very complex solution using a suite of technologies that they aren't used to?
+  - Example: How many ops teams have recently been expected to run a production Kubernetes cluster, even though they might be predominantly experienced with running Windows workloads?
+  - Or, how many dev teams have been told to run private endpoints when they have no networking experience?
+  - Consider the tradeoffs between IaaS, PaaS, and serverless - often the service cost is higher for managed services, but the complexity and your staff cost is lower.
+
+- Consider DevOps requirements. Remember DevOps = Dev + Ops. Don't restrict DevOps to just deployment. For example, consider monitoring.
+
+- Monitoring should also be driven by your requirements. If you have performance requirements to say "end to end transaction latency should be X", how will you actually monitor this? Who will look at the dashboards, and how frequently?
+
+## Cost
+
+- **Pillar Concept** - solution budget, scalable architectures, training/upskilling, cost management
+
+- You need to consider what cost you're prepared to bear.
+- Understand how scalable your architecture should be - e.g. business workload peak times, expected growth.
+- Is this solution a cost center or profit center? ( important for trade-offs)
+
+- Cost includes:
+   - Your Azure costs 
+   - Networking - Egress scenarios. Vnet peering, zone transfer 
+   - Staff costs.
+   - And all of the ancillary costs - e.g. if you have a managed provider, how much are they charging?
+- When you're considering wildly different architectures, consider the total cost. For example, IaaS vs. PaaS architectures have totally different cost profiles.
+
+- Trade offs
+  - Cost vs Uptime
+  - Cost in relation to complexity and operations 
+
+## Wrap-up
+
+- Don't make assumptions about requirements. Get as explicit requirements as possible.
+- We (FTA) can only help you if we understand your requirements. If you don't have them, or if you just make them up on the spot, we can't be as much use to you.
+  - If you don't understand something, ask - we'll spend as much time as we need to help you understand what you need to decide.
+- Remember there are tradeoffs. Don't automatically dial everything up to 11 ("most secure", "most resilient") without thinking through the consequences.
+- **NFRs should come from product owners and business stakeholders, not from the technology builders.** Bring in the right people to help you determine your requirements.
```

Comment:
Agree! But as I mentioned above, technologists should make sure that business stakeholders understand exactly what they are asking for and what it means.

Reply:
Good point - we will revise and emphasise this point.

Reply:
Updated!